src/test/mir-opt/early_otherwise_branch_68867.rs

```diff
@@ -0,0 +1,31 @@
+// compile-flags: -Z mir-opt-level=3
+
+// example from #68867
+type CSSFloat = f32;
+
+pub enum ViewportPercentageLength {
+    Vw(CSSFloat),
+    Vh(CSSFloat),
+    Vmin(CSSFloat),
+    Vmax(CSSFloat),
+}
+
+// EMIT_MIR early_otherwise_branch_68867.try_sum.EarlyOtherwiseBranch.diff
```

Comment:
I think here it would also be very interesting to get a diff from `EarlyOtherwiseBranch` all the way to `SimplifyBranches` after copy prop

Reply:
I pushed a new commit that adds this (if I understood correctly what you meant)

Reply:
Ah, no what I meant was a diff between two passes:

```
// EMIT_MIR early_otherwise_branch_68867.try_sum.EarlyOtherwiseBranch.before.mir early_otherwise_branch_68867.try_sum.SimplifyBranches-after-copy-prop.after.mir
```

Reply:
Rebasing on master, the diff looks even better as `DestinationPropagation` is now in master

Reply:
But looking at the end result, it doesn't look like we are deduplicating the branches' basic blocks. I guess we need to make branch deduplication smarter for that. We need to teach it that different locals don't mean different logic if the locals are only ever used within that basic block

Reply:
Can you change the diff so it shows the full multi-optimization diff?

Reply:
Just pushed a diff from before EarlyOtherwiseBranch to after SimplifyBranches-after-copy-prop. Or do you mean something else?

Reply:
nope, we just had a race condition :D